### PR TITLE
Add Glayout checks in CI

### PR DESCRIPTION
**Pull request by chetanyagoyal**
This PR aims to enable CI checking for glayout. The vision is to enable DRC and LVS checks for the primitives and the components marked with the `@cell` decorator in `glayout/components`. This envolves the following changes:

- [x] Create DRC script for glayout
- [x] Create wrapper script to test all components for DRC errors
- [x] Create `.yml` file for GitHub actions
- [x] Modify LVS script (specifically for glayout)
- [x] Create wrapper script to test all components for LVS errors

### CODE REVIEW

**Comment by chetanyagoyal** [comment]
@alibillalhammoud @andylithia can you help out with a review?

**Comment by chetanyagoyal** [comment]
@msaligane all checks have been added, but this shouldn't be merged until gf180 is added to docker image (required for gf180 drc run)

File: .github/scripts/run_glayout_lvs.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        str: the modified netlist string
+    """
+    pattern_diff = re.compile(r'\bDIFF_PAIR\b')
+    pattern_nmos = re.compile(r'\bNMOS\b')
+    pattern_pmos = re.compile(r'\bPMOS\b')
+    pattern_opamp = re.compile(r'\bopamp\b')
+    patterns = [pattern_diff, pattern_nmos, pattern_pmos, pattern_opamp]
+    replacements = ['diff_test', 'nmos_test', 'pmos_test', 'opamp_test']
+
+    for i, pattern in enumerate(patterns):
```
Comment (alibillalhammoud) [inline .github/scripts/run_glayout_lvs.py:60]: do list comprehension here

File: openfasoc/common/drc-lvs-check/run_lvspex.sh [hunk 2]
```diff
@@ -8,13 +8,41 @@ gds flatglob *\$\$*
...
 
-if {![string compare $2 "ldoInst"]} {
+if {![string compare $2 "diff_test"]} {
   select top cell
-  flatten ldoInst_flat
-  load ldoInst_flat
-  cellname delete ldoInst
-  cellname rename ldoInst_flat ldoInst
+  flatten diff_test_flat
+  load diff_test_flat
```
Comment (alibillalhammoud) [inline openfasoc/common/drc-lvs-check/run_lvspex.sh:14]: Make the first thing LDO, may break the generators

File: openfasoc/common/drc-lvs-check/custom_block_tool/block_lvs_check_sky130pdk/Makefile [hunk 3]
```diff
@@ -1,12 +1,14 @@
 #---------need to be modified by user--------------------------------------------
 # the path for your gds file
-GDS_FILE_PATH = /home/elonjia/Desktop/tool_package/ALIGN-public/work/DCDC_MUX/DCDC_MUX.gds
+GDS_FILE_PATH = /home/chetanya/OpenFASOC/openfasoc/generators/gdsfactory-gen/diff_test.gds
+# GDS_FILE_PATH = /home/elonjia/Desktop/tool_package/ALIGN-public/work/DCDC_MUX/DCDC_MUX.gds
 #the path for your cdl/ngspice file
-CDL_FILE_PATH = /home/elonjia/Desktop/tool_package/ALIGN-public/work/DCDC_MUX/DCDC_MUX.sp
+CDL_FILE_PATH = /home/chetanya/OpenFASOC/openfasoc/common/platforms/sky130hd/cdl/sky130_fd_sc_hd.spice
```
Comment (alibillalhammoud) [inline openfasoc/common/drc-lvs-check/custom_block_tool/block_lvs_check_sky130pdk/Makefile:6]: dont hardcode paths

File: docs/source/notebooks/glayout/glayout_opamp.ipynb [hunk 4]
```diff
@@ -790,7 +790,7 @@
     },
     "language_info": {
       "name": "python",
-      "version": "3.9.13"
+      "version": "3.8.10"
```
Comment (alibillalhammoud) [inline docs/source/notebooks/glayout/glayout_opamp.ipynb:793]: revert this

**Review by alibillalhammoud — CHANGES_REQUESTED** [review]
See reviews

File: docs/source/notebooks/glayout/glayout_opamp.ipynb (hunk 4 above)
Comment (alibillalhammoud) [reply docs/source/notebooks/glayout/glayout_opamp.ipynb:793]: current notebook is known to work

**Comment by chetanyagoyal** [comment]
this should not be merged yet, many changes need to be made

**Pull request closed without merging**